src/core/evaluator.js

```diff
@@ -724,12 +724,14 @@ class PartialEvaluator {
       return;
     }
 
-    const softMask = dict.get("SM", "SMask") || false;
-    const mask = dict.get("Mask") || false;
-
     const SMALL_IMAGE_DIMENSIONS = 200;
     // Inlining small images into the queue as RGB data
-    if (isInline && !softMask && !mask && w + h < SMALL_IMAGE_DIMENSIONS) {
+    if (
+      isInline &&
+      !dict.has("SMask") &&
+      !dict.has("Mask") &&
+      w + h < SMALL_IMAGE_DIMENSIONS
```

Comment:
Nit: if it's a hot path, might be worth moving the last check after isInline, as it's likely faster than the `has` check

Reply:
This really shouldn't matter, performance wise, since:
 - This code only applies when parsing *inline* images, and those are fortunately not too common.
 - We've got multiple levels of image-caches on the worker-thread.
 - The main point of this patch was to avoid the unnecessary Stream fetching/parsing.